### ISSUE TITLE
no-empty-interface should not fail on an interface with multiple parents

### ISSUE THREAD

**Issue by HamletDRC** [palantir/tslint#2065]
### Bug Report

- __TSLint version__: `4.3.0`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: grunt tslint

#### TypeScript code being linted

```ts
            interface MyInterface extends First, Second {
            }
```

An interface with two parents is not empty. It is a common way to merge disparate attributes into a meaningful composition. 

For example, we use this by having three interfaces: HasWidth, HasHeight, and a 3rd called HasDimensions that implements the first two. Then there are functions that operate on the appropriate abstraction.

**Comment by mohsen1** [comment]
Isn't it cleaner to use intersection to merge those types instead of extending them with an empty Interface?

**Comment by mohsen1** [comment]
```to
type MyInterface = First & Second 
```

**Comment by HamletDRC** [comment]
Whether or not an alternative is cleaner does not change the fact that an interface with two parents is not empty. The preference for intersection types that you describe may be a valid preference, but it is a different tslint rule that should enforce that.
